### PR TITLE
fix: Adding build tag for Rosetta

### PR DESCRIPTION
**Pull request by hacheigriega**
## Motivation

Adding an optional build tag called `rosetta` to determine whether to add Rosetta support or not.
To build with this tag using make, run:
```
ENABLE_ROSETTA=true make build
```

Related: #298 
Thank you @Thomasvdam for the solution!

### CODE REVIEW

**Review by Thomasvdam — APPROVED** [review]
It feels a little clunky but I think this is the way to Go 👉 😎 👉

**Comment by Thomasvdam** [comment]
Actually we might want to add a note somewhere that rosetta and static can't go together 🤔 Maybe in the Makefile and have it error when both tags are present?

Or we add require the static tag to not be set for the Rosetta file, I think you can do boolean logic with tags.

File: cmd/sedad/cmd/rosetta_disabled.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//go:build !rosetta
```
Comment (Thomasvdam) [inline cmd/sedad/cmd/rosetta_disabled.go:1]: We can make this

```golang
//go:build !rosetta static
```

And the other

```golang
//go:build rosetta,!static
```

Provided there is a static build tag... 😬

**Comment by hacheigriega** [comment]
Added an error message when both `ENABLE_ROSETTA` and `LINK_STATICALLY` are set to true.
```
LINK_STATICALLY=true ENABLE_ROSETTA=true make build                                                      
Makefile:78: *** Cannot link statically when Rosetta is enabled.  Stop.
```

**Review by gluax — APPROVED** [review]

**Pull request merged**